tests/integration/main.cpp

```
@@ -0,0 +1,32 @@
+
+int main()
+{
+    LSODA solver;
+    std::vector<double> y = { 1.0 }, yout;
+    double t = 0.0;
+    int istate = 1;
+
+    solver.lsoda_update(exponential_decay, 1, y, yout, &t, 1.0, &istate, nullptr, 1e-8, 1e-10);
+
+    if (istate <= 0)
+        throw std::runtime_error("lsoda_update failed: istate=" + std::to_string(istate));
```

Cppcheck (2.20.0)

[error] Unhandled exception thrown in function that is an entry point.

(throwInEntryPoint)